Source/Core/Core/HW/DSPHLE/DSPHLE.cpp

```diff
@@ -191,26 +191,37 @@ void DSPHLE::DSP_WriteMailBoxLow(bool cpu_mailbox, u16 value)
 // Other DSP functions
 u16 DSPHLE::DSP_WriteControlRegister(u16 value)
 {
-  DSP::UDSPControl temp(value);
```

Comment:
Okay, just to clarify, you'd like to see DSP_ReadControlRegister changed to:

```c++
u16 DSPHLE::DSP_ReadControlRegister()
{
  u32 crc = UCodeInterface::GetCRC(m_ucode.get());
  u16 control = m_dsp_control.Hex;
  if (crc == UCODE_ROM || crc == UCODE_INIT_AUDIO_SYSTEM)
    control |= 0x800;
  else
    control &= ~0x800;
  return control;
}
```

Is that correct?